edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -91,9 +91,14 @@ public virtual async Task<Stream> GetModuleLogs(string module, bool follow, Opti
                 string baseUrl = HttpClientHelper.GetBaseUrl(this.ManagementUri).TrimEnd('/');
                 var logsUrl = new StringBuilder();
                 logsUrl.AppendFormat(CultureInfo.InvariantCulture, LogsUrlTemplate, baseUrl, module, this.Version.Name, follow.ToString().ToLowerInvariant());
-                tail.ForEach(t => logsUrl.AppendFormat($"&{LogsUrlTailParameter}={t}"));
                 since.ForEach(s => logsUrl.AppendFormat($"&{LogsUrlSinceParameter}={Uri.EscapeUriString(s)}"));
                 until.ForEach(u => logsUrl.AppendFormat($"&{LogsUrlUntilParameter}={Uri.EscapeUriString(u)}"));
+
+                if (!(tail.HasValue && since.HasValue && until.HasValue))
```

Comment:
Yup, absolutely. This is where we about to send the docker API request via mgmt.sock. 
Starting from this PR, the behavior is separated into 2 scenarios: 
1. When all 3 options (tail + since + until) are specified, only the `since + until` is passed down to mgmt.sock 
2. When other combination of options (tail+since, tail+until, until+since) are specified, the options are then passthrough edgeAgent into mgmt.sock directly. 

Once the log from (1) is returned, the result is `tail`-ed after Akka parsing to enforce the tail.